cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,267 @@
+        """
+        # Log out then try to retrieve the page
+        self.client.logout()
+        response = self.client.get(url)
+
+        # Expect a redirect to the login page
+        redirect_url = '{login_url}?next={original_url}'.format(
+            login_url=reverse('login'),
+            original_url=url,
+        )
+
+        self.assertRedirects(response, redirect_url)
+
+    @ddt.data(MAINTENANCE_URLS)
+    @ddt.unpack
+    def test_global_staff_access(self, url):
+        """
+        Test that all maintenance app views are accessible to global staff user.
+        """
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+    @ddt.data(MAINTENANCE_URLS)
+    @ddt.unpack
+    def test_non_global_staff_access(self, url):
+        """
+        Test that all maintenance app views are not accessible to non-global-staff user.
+        """
+        user = UserFactory(username='test', email='test@example.com', password='test')
+        login_success = self.client.login(username=user.username, password='test')
+        self.assertTrue(login_success)
+
+        response = self.client.get(url)
+        self.assertContains(response, 'Must be edX staff to perform this action.', status_code=403)
+
+
+@ddt.ddt
+class TestForcePublish(MaintenanceViewTestCase):
+    """
+    Tests for the force publish view.
+    """
+
+    def setUp(self):
+        super(TestForcePublish, self).setUp()
+        self.view_url = reverse('maintenance:force_publish_course')
+
+    def setup_test_course(self):
+        """
+        Creates the course and add some changes to it.
+
+        Returns:
+            course: a course object
+        """
+        course = CourseFactory.create(default_store=ModuleStoreEnum.Type.split)
+        # Add some changes to course
+        chapter = ItemFactory.create(category='chapter', parent_location=course.location)
+        self.store.create_child(
+            self.user.id,  # pylint: disable=no-member
+            chapter.location,
+            'html',
+            block_id='html_component'
+        )
+        # verify that course has changes.
+        self.assertTrue(self.store.has_changes(self.store.get_item(course.location)))
+        return course
+
+    @ddt.data(
+        ('', COURSE_KEY_ERROR_MESSAGES['empty_course_key']),
+        ('edx', COURSE_KEY_ERROR_MESSAGES['invalid_course_key']),
+        ('course-v1:e+d+X', COURSE_KEY_ERROR_MESSAGES['course_key_not_found']),
+    )
+    @ddt.unpack
+    def test_invalid_course_key_messages(self, course_key, error_message):
+        """
+        Test all error messages for invalid course keys.
+        """
+        # validate that course key contains error message
+        self.verify_error_message(
+            data={'course-id': course_key},
+            error_message=error_message
+        )
+
+    def test_non_split_course(self):
+        """
+        Test that we get a error message on non-split courses.
```

Comment:
As we are specifically testing `mongo` modulestore, I think this should be `... error message on mongo courses.`